storage/mysql/log_storage.go

```diff
@@ -569,32 +568,21 @@ func (t *logTreeTX) UpdateSequencedLeaves(ctx context.Context, leaves []*trillia
 
 // removeSequencedLeaves removes the passed in leaves slice (which may be
 // modified as part of the operation).
-func (t *logTreeTX) removeSequencedLeaves(ctx context.Context, leaves []*trillian.LogLeaf) error {
-	// Delete in order of the hash values in the leaves.
-	sort.Sort(byLeafIdentityHash(leaves))
-
+func (t *logTreeTX) removeSequencedLeaves(ctx context.Context, leaves []*dequeuedLeaf) error {
```

Comment:
It's kind of ambiguous but I don't think it's wrong it's removing leaves from the unsequenced table that have just been sequenced.